llama_agents/orchestrators/agent.py

```
@@ -160,16 +160,17 @@ async def add_result_to_state(
         # summarize the result
         new_history = result.history
         new_history_str = "\n".join([str(x) for x in new_history])
-        # TODO: Better logic for when to summarize?
-        if len(new_history) > 1:
+
+        summary = None
+        if self.should_summarize(new_history):
             summarize_prompt_str = self.summarize_prompt.format(history=new_history_str)
             summary = await self.llm.acomplete(summarize_prompt_str)
 
         # get the current chat history, add the summary to it
         chat_dicts = state.get(HISTORY_KEY, [])
         chat_history = [ChatMessage(**x) for x in chat_dicts]
-
-        chat_history.append(ChatMessage(role="assistant", content=str(summary)))
```

I guess this should throw an undefined error if new_history had length 0 or 1?

---

Let me know if the way I handle this in  `llama_agents/control_plane/server.py` makes sense. I use `ValueError` though or maybe instead of Undefined a custom `EmptyTaskHistoryError` ?